File: src/data/crawler/web/LookupSplit.scala
package data.crawler.web

import java.util

class LookupSplit(ttype: String, llabal: String, splitRegex: String) extends LookupPattern(ttype, llabal, null) {

  override protected def getResults(propertyMap: util.Map[String, String], partial: String): util.List[String] = {

    val array = partial.split(splitRegex)
    if (isNth && nth < array.length) {
      var resultList = new util.ArrayList[String]()
      var i = nth
      while (i < mth) {
        val result = getReplaces(array(i))
        resultList.add(result)
        i += 1
      }

      resultList
    }
    else {
      util.Arrays.asList[String](array: _*)
    }

  }
}

File: src/data/crawler/web/LookupSplit.java
package data.crawler.web;


import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

public class LookupSplit extends LookupPattern {
    String splitRegex;

    public LookupSplit(String type, String label, String splitRegex) {
        super(type, label, null);

        this.splitRegex = splitRegex;
    }

    public List<String> getResults(Map<String, String> propertyMap, String partial) {

        String[] array = partial.split(splitRegex);
        if (isNth() && nth < array.length) {
            List<String> resultList = new ArrayList<String>();
            int i = nth;
            while (i < mth) {
                String result = getReplaces(array[i]);
                resultList.add(result);
                i += 1;
            }

            return resultList;
        } else {
            return Arrays.asList(array);
        }

    }
}
